File: src/leetcode/array/ContainerWithMostWater.java
package leetcode.array;

public class ContainerWithMostWater {
    class Solution {
        public int maxArea(int[] height) {
            int n = height.length;
            int answer = 0;
            for(int l = 0, r = n - 1; l < r; ) {

                int area = (r - l) * Math.min(height[l], height[r]);
                answer = Math.max(area, answer);

                if(height[l] > height[r]) r--;
                else l++;
            }

            return answer;
        }
    }
}

File: src/leetcode/array/ContainerWithMostWater.js
var maxArea = function(height) {
    const n = height.length;
    let max = 0;
    for (let left = 0, right = n - 1; left < right; ) {

        const area = (right - left) * Math.min(height[right], height[left]);
        max = Math.max(area, max);

        if(height[left] > height[right]) right--;
        else left++;
    }

    return max;
};